.github/workflows/lint.yml

```
@@ -40,21 +40,20 @@ jobs:
       - name: Install Node.js dependencies
         run: npm ci
 
-      - name: Build App
-        run:  npm run build
+      - name: Start Container
+        run: docker-compose -f "docker-compose.yml" up -d
 
       - name: Run linters
         uses: wearerequired/lint-action@v2
         with:
           eslint: true
-          eslint_extensions: "js,jsx,ts,tsx"
-
-      - name: Start Container
-        run: docker-compose -f "docker-compose.yml" up -d
 
-      - name: Run Tests
+      - name: Run Jest tests
         run: npm run test
 
+      - name: Cypress run
+        uses: cypress-io/github-action@v5.1.0 # use the explicit version number
+
```

Looking good, but you need to build and start the server... something like this should get you going:
```suggestion
      - name: Cypress run
        uses: cypress-io/github-action@v5.1.0 # use the explicit version number
          build: npm run build
          start: npm start
          wait-on: 'http://localhost:3000'
```

There's lots of useful information on [cypress-io/github-action](https://github.com/cypress-io/github-action). 👍🏼

---

I've both npm run build and npm start are part of the dockerfile, and run when it's called by the docker compose file. do I have to have to repeat the statements again?

---

I'm uncertain as to why a Dockerfile is being used. Was there a particular need for it?

If we keep it, we'll need to edit the `npm` scripts to only start the `pg` service or create a separate docker-compose for the workflow.

If it's not needed and doesn't provide any additional benefit, I'd opt to do away with it and stick to the changes I suggested above.

---

ideally we'd use the built files for other things as well, such as running server tests.  i imagine we _could_ use the cypress build step for this but i think it'd be better to have the different test suites to work independently of each other, even if we can guarantee that we'll always run jest acceptance tests after cypress e2e tests.

---

Fair... In that case, I guess we do need a second `docker-compose` file... one with the app build step.

We'll also need to add an environment variable for `DATABASE_URL` with the host configured as `pg`.